File: cpp_files/raylib/basic_window_resizable.cpp
#include <algorithm>
#include <cassert>
#include <raylib.h>

constexpr int BASE_WIDTH  = 800;
constexpr int BASE_HEIGHT = 600;

int main(void) {
    SetConfigFlags(FLAG_WINDOW_RESIZABLE);
    InitWindow(BASE_WIDTH, BASE_HEIGHT, "");
    SetTargetFPS(60);

    RenderTexture canvas = LoadRenderTexture(BASE_WIDTH, BASE_HEIGHT);
    assert(IsRenderTextureValid(canvas));

    while (!WindowShouldClose()) {
        BeginTextureMode(canvas);
        ClearBackground(RAYWHITE);
        EndTextureMode();

        const float scale =
            std::min(static_cast<float>(GetScreenWidth()) / static_cast<float>(BASE_WIDTH),
                     static_cast<float>(GetScreenHeight()) / static_cast<float>(BASE_HEIGHT));
        const Vector2 offset = {
            (GetScreenWidth() - (BASE_WIDTH * scale)) / 2.0f,
            (GetScreenHeight() - (BASE_HEIGHT * scale)) / 2.0f,
        };

        const Rectangle source = {0, 0, BASE_WIDTH, BASE_HEIGHT};
        const Rectangle dest   = {offset.x, offset.y, BASE_WIDTH * scale, BASE_HEIGHT * scale};

        BeginDrawing();
        ClearBackground(BLACK);

        DrawTexturePro(canvas.texture, source, dest, {0, 0}, 0.0f, WHITE);

        EndDrawing();
    }

    UnloadRenderTexture(canvas);

    CloseWindow();

    return 0;
}

File: python_files/raylib_files/basic_window_resizable.py
import pyray as p

SCREEN_WIDTH: int = 800
SCREEN_HEIGHT: int = 600


def main() -> None:
    p.set_config_flags(p.ConfigFlags.FLAG_WINDOW_RESIZABLE)

    p.init_window(SCREEN_WIDTH, SCREEN_HEIGHT, "")
    p.set_target_fps(60)

    canvas: p.RenderTexture = p.load_render_texture(SCREEN_WIDTH, SCREEN_HEIGHT)
    assert p.is_render_texture_valid(canvas)

    while not p.window_should_close():
        p.begin_texture_mode(canvas)
        p.clear_background(p.RAYWHITE)
        p.end_texture_mode()

        scale: float = min(
            p.get_screen_width() / SCREEN_WIDTH, p.get_screen_height() / SCREEN_HEIGHT
        )
        offset: p.Vector2 = p.Vector2(
            (p.get_screen_width() - (SCREEN_WIDTH * scale)) / 2,
            (p.get_screen_height() - (SCREEN_HEIGHT * scale)) / 2,
        )

        source: p.Rectangle = p.Rectangle(0, 0, SCREEN_WIDTH, -SCREEN_HEIGHT)
        dest: p.Rectangle = p.Rectangle(
            offset.x, offset.y, SCREEN_WIDTH * scale, SCREEN_HEIGHT * scale
        )

        p.begin_drawing()
        p.clear_background(p.BLACK)

        p.draw_texture_pro(canvas.texture, source, dest, p.Vector2(0, 0), 0.0, p.WHITE)

        p.end_drawing()

    p.unload_render_texture(canvas)

    p.close_window()


if __name__ == "__main__":
    main()
